proxmoxtf/resource/container/container.go

```diff
@@ -2343,85 +2329,56 @@ func containerRead(ctx context.Context, d *schema.ResourceData, m interface{}) d
 		initialization[mkInitializationHostname] = ""
 	}
 
-	devicePassthroughArray := []*containers.CustomDevicePassthrough{
-		containerConfig.DevicePassthrough0,
-		containerConfig.DevicePassthrough1,
-		containerConfig.DevicePassthrough2,
-		containerConfig.DevicePassthrough3,
-		containerConfig.DevicePassthrough4,
-		containerConfig.DevicePassthrough5,
-		containerConfig.DevicePassthrough6,
-		containerConfig.DevicePassthrough7,
-	}
-
-	devicePassthroughList := make([]interface{}, 0, len(devicePassthroughArray))
-
-	for _, dp := range devicePassthroughArray {
-		if dp == nil {
-			continue
-		}
+	passthroughDevicesMap := make(map[string]interface{}, len(containerConfig.PassthroughDevices))
 
-		devicePassthrough := map[string]interface{}{}
+	for key, dp := range containerConfig.PassthroughDevices {
+		passthroughDevice := map[string]interface{}{}
 
 		if dp.DenyWrite != nil {
-			devicePassthrough[mkDevicePassthroughDenyWrite] = *dp.DenyWrite
+			passthroughDevice[mkDevicePassthroughDenyWrite] = *dp.DenyWrite
 		} else {
-			devicePassthrough[mkDevicePassthroughDenyWrite] = false
+			passthroughDevice[mkDevicePassthroughDenyWrite] = false
 		}
 
 		if dp.GID != nil {
-			devicePassthrough[mkDevicePassthroughGID] = *dp.GID
+			passthroughDevice[mkDevicePassthroughGID] = *dp.GID
 		} else {
-			devicePassthrough[mkDevicePassthroughGID] = 0
+			passthroughDevice[mkDevicePassthroughGID] = 0
 		}
 
 		if dp.Mode != nil {
-			devicePassthrough[mkDevicePassthroughMode] = *dp.Mode
+			passthroughDevice[mkDevicePassthroughMode] = *dp.Mode
 		} else {
-			devicePassthrough[mkDevicePassthroughMode] = dvDevicePassthroughMode
+			passthroughDevice[mkDevicePassthroughMode] = dvDevicePassthroughMode
 		}
 
-		devicePassthrough[mkDevicePassthroughPath] = dp.Path
+		passthroughDevice[mkDevicePassthroughPath] = dp.Path
 
 		if dp.UID != nil {
-			devicePassthrough[mkDevicePassthroughUID] = *dp.UID
+			passthroughDevice[mkDevicePassthroughUID] = *dp.UID
 		} else {
-			devicePassthrough[mkDevicePassthroughUID] = 0
+			passthroughDevice[mkDevicePassthroughUID] = 0
 		}
 
-		devicePassthroughList = append(devicePassthroughList, devicePassthrough)
+		passthroughDevicesMap[key] = passthroughDevice
 	}
 
-	currentDevicePassthrough := d.Get(mkDevicePassthrough).([]interface{})
+	passthroughDevices := utils.OrderedListFromMap(passthroughDevicesMap)
```

Comment:
That might not work as expected, as keys in `OrderedListFromMap` are sorted alphabetically. So `dev11` will come before `dev2`

Reply:
fixed